README.md

```diff
@@ -42,15 +121,99 @@ Install requirements:
 npm install
 ```
 
-## Getting started
+#### Setup environments
 
-### How to make a deposit
+Being a NPM package, the SDK itself doesn’t use environment variables with exception of the tests. But in order to use the SDK you will need to set up an environment file following a specific .env.example file.
+Env.example
 
-```bash
-npm run-script eg:deposit
+##### Locally (Ganache)
+
+Create .env.ganache file
+
+```
+APP_BLOCKCHAIN_WEBSOCKET_URL=ws://localhost:8546
+APP_CLIENT_API_URL=http://localhost:8080
+APP_ETH_PRIVATE_KEY=0x4775af73d6dc84a0ae76f8726bda4b9ecf187c377229cb39e1afa7a18236a69e
+APP_NIGHTFALL_MNEMONIC=
+APP_TOKEN_ADDRESS_1=0x4f3c4F8D4575Cf73c2FAf9F36cc505e19E65B9C0
 ```
-WIP, explain env file
 
+##### Goerli Testnet
+
+Create .env.goerli file
+
+```
+APP_BLOCKCHAIN_WEBSOCKET_URL=wss://goerli.infura.io/ws/v3/89114afb34fb4dc29d333f9849679c53
+APP_CLIENT_API_URL=http://localhost:8080
+APP_ETH_PRIVATE_KEY=
+APP_NIGHTFALL_MNEMONIC=
+APP_TOKEN_ADDRESS_1=0x499d11E0b6eAC7c0593d8Fb292DCBbF815Fb29Ae
+```
+
+##### EthereumMainnet
+
+### Getting started
+
+To get a good idea of how to interact with Nightfall using the SDK, there are a set of example scripts that allow for a better understanding of the core features of Nightfall.
+
+All of the scripts are explained in short detail below and are to be run in a local environment.
+
+\*Note
+The scripts load an .env file, it is suggested to use different files per environment, for example creating `examples/.env.ganache` for a local environment, based on `.env.example`.
+Upon each running of any of the scripts a new instance of the User class is created, if an existing Mnemonic isn’t specified in the `.env` file, a new Mnemonic is assigned to that specific user.
+
+#### Make a deposit
+
+```
+npm run-script eg:deposit:ganache
+```
+
+#### Make a transfer
+
+For creating a transfer an already existing account with deposit is required. For testing purposes, this can be achieved by saving the mnemonic used for previous deposits and adding it to the .env file. Two deposits are needed to be able to make a transfer.
+
+```
+npm run-script eg:transfer:ganache
+```
+
+#### Make a withdrawal
+
+To be able to withdraw a balance in Nightfall is needed, this means that at least two deposits are to be made for the proposer to make a block and settle these deposits.
+
+```
+npm run-script eg:withdrawal:ganache
+```
+
+#### Finalise withdrawal
+
+After initiating a withdrawal, two transactions need to be made in order for the proposer to create a block. Upon creating a withdrawal request, finalising the withdrawal means getting the funds back to L1
+
+```
+npm run-script eg:finalise-withdrawal:ganache
+```
+
+#### Check L2 balances
+
+```
+npm run-script eg:nf-balances:ganache
+```
+
+#### Export commitments
+
+```
+npm run-script eg:export-commitments:ganache
+```
+
+#### Import commitments
+
+```
+npm run-script eg:nf-balances
+```
+
+### Eth testing account and tokens
```

Comment:
Maybe we can remove this for now, if we don't have time. Not sure who added this, but probably was for explaining a bit about MetaMask and faucets. We can also check the Polygon docs. Possibly there's something there and we can avoid repetition.